embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/execution/OkHttpRequestExecutionService.kt

```diff
@@ -41,44 +38,33 @@ class OkHttpRequestExecutionService(
     override fun attemptHttpRequest(
         payloadStream: () -> InputStream,
         envelopeType: SupportedEnvelopeType,
-    ): ApiResponse {
+    ): ExecutionResult {
         val apiRequest = envelopeType.endpoint.getApiRequestFromEndpoint()
-
         val requestBody = generateRequestBody(payloadStream)
-
         val request = Request.Builder()
             .url(apiRequest.url)
             .headers(apiRequest.getHeaders().toHeaders())
             .post(requestBody)
             .build()
 
+        var failureReason: Throwable? = null
         val httpCallResponse = try {
             okHttpClient.newCall(request).execute()
         } catch (throwable: Throwable) {
-            return ApiResponse.Incomplete(throwable)
+            failureReason = throwable
+            null
         }
 
-        httpCallResponse.use { response ->
-            return when (response.code) {
-                HttpURLConnection.HTTP_OK -> return ApiResponse.Success(null, null)
-                HttpURLConnection.HTTP_NOT_MODIFIED -> ApiResponse.NotModified
-                HttpURLConnection.HTTP_ENTITY_TOO_LARGE -> ApiResponse.PayloadTooLarge
-
-                TOO_MANY_REQUESTS -> {
-                    val retryAfter = response.headers["Retry-After"]?.toLongOrNull()
-                    ApiResponse.TooManyRequests(envelopeType.endpoint, retryAfter)
-                }
-
-                NO_HTTP_RESPONSE -> ApiResponse.Incomplete(
-                    IllegalStateException("Connection failed or unexpected response code")
-                )
-
-                else -> ApiResponse.Failure(
-                    response.code,
-                    response.headers.associate { it.first to it.second }
-                )
-            }
-        }
+        return getResult(
+            endpoint = envelopeType.endpoint,
+            responseCode = httpCallResponse?.code,
+            headersProvider = {
+                httpCallResponse?.run {
```

Comment:
is the `run` clause needed?
```
httpCallResponse?.headers?.toMap() ?: emptyMap()
```

Reply:
haha good catch. no, I refactored this so yeah no longer needed. Will remove